### ISSUE TITLE
Go to definition not working for external files

### ISSUE THREAD

**Issue by CRuebel** [spyder-ide/spyder#3996]
## Description

**What steps will reproduce the problem?**

1. Select the instance of a class / object that depends on another file somewhere and right click
2. Select "go to definition" from drop down menu
3. 

**What is the expected output? What do you see instead?**
If object was defined in the document that is open, it will go to the definition.  If selecting something that relies on a separate document (something in the Python library), nothing will happen.

**Please provide any additional information below**
Spyder was installed through the Anaconda distribution.

## Version and main components

* Spyder Version: 3.0.2 
* Python Version: 3.5.2
* Qt Versions: 5.6.0, PyQt5 5.6 on Windows

## Dependencies
```
jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.3 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.11.1 (OK)
pandas >=0.13.1  :  0.18.1 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.6.0 :  1.3.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.5.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.4.6 (OK)
sympy >=0.7.3    :  1.0 (OK)

```

**Comment by sroodhorst** [comment]
Having the same problem here. When I link to a module like: 
`from cnntools import code-snippets as code`
`code.printsomething`

when I want to link to the definition of code in _cnntools_, which is another project folder in the parent directory, spyder is not doing anything.

Currently using Python 3.5.2 64 bits, Qt 4.8.7, PyQt 4.11.4, spyder 3.1.3, Windows 7 64bits.
Spyder dependencies all up to date.

```
jedi =0.9.0      :  0.9.0 (OK)
matplotlib >=1.0 :  2.0.0 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.12.1 (OK)
pandas >=0.13.1  :  0.19.2 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.5.1 (OK)
sympy >=0.7.3    :  1.0 (OK)
```


Is there any news on this issue?

**Comment by ccordoba12** [comment]
You need to add your projects to our Python path manager for this to work. However, there are some errors with this approach (reported in issue #4410) that will be addressed in our **3.2.2** version.

For for the inconvenience in the meantime.

----

*Note*: You can find our Python path manager in the menu

    Tools > PYTHONPATH manager
